### ISSUE TITLE
use 'safe' as variable name generate wrong diagnostic

### ISSUE THREAD

**Issue by fda-odoo** [rust-lang/rust-analyzer#18445]
**rust-analyzer version**: 0.3.2154-standalone (9323b5385 2024-10-20)

**rustc version**: rustc 1.81.0 (eeb90cda1 2024-09-04)

**editor or extension**: Vscode - Extension "rust-analyzer" version v0.3.2162

**code snippet to reproduce**: 
```rust
        let mut safe = false;
        safe = true;

```

Even if this code compile (safe is a situational keyword), rust-analyzer is expecting a function after the keyword safe and generate an error:
"Syntax Error: expected fn, trait or impl"

**Comment by ChayimFriedman2** [comment]
Oh it's because `safe` is marked contextual before `fn`, but `fn` can appear in statement position, except `safe` can only appear in `extern` blocks...

@rustbot claim
